namer/consul/src/main/scala/io/buoyant/namer/consul/LookupCache.scala

```diff
@@ -41,13 +42,21 @@ private[consul] class LookupCache(
     }
   }
 
-  // Note that this activity is never recomputed.  We simply do a
+  private[this] def resolveDc(Datacenter: String): Activity[String] = localDc match {
+    case Some(Datacenter) => config.map(_.flatMap(_.Datacenter).getOrElse(Datacenter))
```

Comment:
I used upper-case to make it stable identifier so I can use it in match but it's really not needed with the approach you proposed - thanks